src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -182,10 +183,28 @@ protected function compileJoins(Builder $query, $joins)
 
             $tableAndNestedJoins = is_null($join->joins) ? $table : '('.$table.$nestedJoins.')';
 
+            if ($join instanceof JoinLateralClause) {
+                return $this->joinLateral($join, $tableAndNestedJoins);
+            }
+
             return trim("{$join->type} join {$tableAndNestedJoins} {$this->compileWheres($join)}");
         })->implode(' ');
     }
 
+    /**
+     * Compile a "lateral join" clause.
+     *
+     * @param  \Illuminate\Database\Query\JoinLateralClause  $join
+     * @param  string  $expression
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function joinLateral(JoinLateralClause $join, string $expression): string
```

Comment:
Can you rename this method to `compileJoinLateral()`? That's more consistent with the rest of the class.